### PR TITLE
Restrict consultant role edit permissions on project fields

### LINKED ISSUE #2185: User can "Edit" the project  even user has "No" update access
**Roles:** Consultant

**Describe the bug**
Project > User can `Edit` the project  even user has `No` update access

**Steps to Reproduce**
1.  Login as a root user
2.  Create a project and open a project detail page
3.  Add team member `Consultant`
4.  Login as a `Consultant` user
5.  Open created a project details page
6. Click on the `Edit` icon and edit the project details

**Expected Result**
User shouldn't be allowed to edit the project when the user has `No` update access
 
**Actual Result**
User can `Edit` the project  even user has `No` update access

See video - 

https://user-images.githubusercontent.com/5212585/135823856-8d72dc94-162d-4ad7-b4de-815f9c24191c.mp4

image

**FYI:** @ParkeBrown @michaelmarshall 

CordAPIV3# c761327
cord-field# 684cfb8

### PR DESCRIPTION
**Pull request by daniel-leod**
Fixes #2185 #2180 #2146

### CODE REVIEW

File: src/components/authorization/roles/consultant-on-project.role.ts [hunk 1]
```diff
@@ -335,31 +335,31 @@ export const ConsultantOnProject = new DbRole({
...
+        { propertyName: 'otherLocations', permission: { read, }, },
+        { propertyName: 'primaryLocation', permission: { read, }, },
+        { propertyName: 'marketingLocation', permission: { }, },
+        { propertyName: 'partnership', permission: { read, }, },
+        { propertyName: 'budget', permission: { }, },
+        { propertyName: 'modifiedAt', permission: { read, }, },
+        { propertyName: 'fieldRegion', permission: { read, }, },
+        { propertyName: 'engagement', permission: { read, }, },
+        { propertyName: 'sensitivity', permission: { read, }, },
         { propertyName: 'tags', permission: { read, write, }, },
```
Comment (elisabeth-sorrell) [inline src/components/authorization/roles/consultant-on-project.role.ts:358]: Just double-checked with Carson and we can go ahead and delete write for `tags`, too

**Review by elisabeth-sorrell — DISMISSED** [review]

File: src/components/authorization/roles/consultant-on-project.role.ts (hunk 1 above)
Comment (daniel-leod) [reply src/components/authorization/roles/consultant-on-project.role.ts:358]: Ok I'm on it

**Comment by bryanjnelson** [comment]
@daniel-leod - Are you sure this will fix #2146 ?  I see that one deals with the Staff Member role as well as consultant.  Since you only modify the consultant-on-project.role.ts here I'm not sure this will fully close that particular issue?

**Comment by daniel-leod** [comment]
> @daniel-leod - Are you sure this will fix #2146 ? I see that one deals with the Staff Member role as well as Consultant. Since you only modify the consultant-on-project.role.ts here I'm not sure this will fully fix that particular issue?

I think [this line](https://github.com/SeedCompany/cord-api-v3/pull/2175/commits/b091f2d07f74115d9b34cb8034bdf00405b0df57#diff-288fb951eb5019b7cfeee636f32a64ea1f260e0a1f73a574fba0c4dc4ac56da7R344) fixes the budget permission issue for staff member role.

**Comment by bryanjnelson** [comment]
@daniel-leod - Correct, but since #2146 also deals with the **Staff Member** role I don't think we should close that issue with this PR.  Maybe just reference it and then make a note in it, but don't actually "fix" it or else it may get prematurely closed.

Also, I have a question out to Seth tomorrow about this and then hopefully I can finish the review!

File: src/components/authorization/roles/consultant-on-project.role.ts [hunk 2]
```diff
@@ -335,31 +335,31 @@ export const ConsultantOnProject = new DbRole({
...
+        { propertyName: 'mouStart', permission: { read, }, },
+        { propertyName: 'mouEnd', permission: { read, }, },
+        { propertyName: 'initialMouEnd', permission: { read, }, },
+        { propertyName: 'stepChangedAt', permission: { read, }, },
+        { propertyName: 'rootDirectory', permission: { read, write }, },
+        { propertyName: 'member', permission: { read, }, },
+        { propertyName: 'otherLocations', permission: { read, }, },
+        { propertyName: 'primaryLocation', permission: { read, }, },
+        { propertyName: 'marketingLocation', permission: { }, },
+        { propertyName: 'partnership', permission: { read, }, },
```
Comment (bryanjnelson) [inline src/components/authorization/roles/consultant-on-project.role.ts:352]: Looks great except the read permissions on `partnership` need locked down a bit more to Low Sensitivity.

See [here for an example](https://github.com/SeedCompany/cord-api-v3/blob/827b5e3d140c4fcfe6c1f0dfcab7f51fefa0e393/src/components/authorization/roles/marketing-on-project.role.ts#L109).

**Review by bryanjnelson — CHANGES_REQUESTED** [review]

File: src/components/authorization/roles/consultant-on-project.role.ts (hunk 2 above)
Comment (daniel-leod) [reply src/components/authorization/roles/consultant-on-project.role.ts:352]: Ok I'm on it

**Review by bryanjnelson — APPROVED** [review]
@daniel-leod - looks good!

@CarsonF - ready for merge, sire.

**Review by CarsonF — APPROVED** [review]
🎉

**Pull request merged**